provider/lpwindow/submit_task.go

```diff
@@ -0,0 +1,272 @@
+	}
+
+	mi, err := w.api.StateMinerInfo(context.Background(), maddr, types.EmptyTSK)
+	if err != nil {
+		return false, xerrors.Errorf("error getting miner info: %w", err)
+	}
+
+	msg := &types.Message{
+		To:     maddr,
+		Method: builtin.MethodsMiner.SubmitWindowedPoSt,
+		Params: pbuf.Bytes(),
+		Value:  big.Zero(),
+
+		From: mi.Worker, // set worker for now for gas estimation
+	}
+
+	// calculate a more frugal estimation; premium is estimated to guarantee
+	// inclusion within 5 tipsets, and fee cap is estimated for inclusion
+	// within 4 tipsets.
+	minGasFeeMsg := *msg
+
+	minGasFeeMsg.GasPremium, err = w.api.GasEstimateGasPremium(context.Background(), 5, msg.From, msg.GasLimit, types.EmptyTSK)
+	if err != nil {
+		log.Errorf("failed to estimate minimum gas premium: %+v", err)
+		minGasFeeMsg.GasPremium = msg.GasPremium
+	}
+
+	minGasFeeMsg.GasFeeCap, err = w.api.GasEstimateFeeCap(context.Background(), &minGasFeeMsg, 4, types.EmptyTSK)
+	if err != nil {
+		log.Errorf("failed to estimate minimum gas fee cap: %+v", err)
+		minGasFeeMsg.GasFeeCap = msg.GasFeeCap
+	}
+
+	// goodFunds = funds needed for optimal inclusion probability.
+	// minFunds  = funds needed for more speculative inclusion probability.
+	goodFunds := big.Add(minGasFeeMsg.RequiredFunds(), minGasFeeMsg.Value)
+	minFunds := big.Min(big.Add(minGasFeeMsg.RequiredFunds(), minGasFeeMsg.Value), goodFunds)
+
+	from, _, err := w.as.AddressFor(context.Background(), w.api, mi, api.PoStAddr, goodFunds, minFunds)
+	if err != nil {
+		return false, xerrors.Errorf("error getting address: %w", err)
+	}
+
+	msg.From = from
+
+	mss := &api.MessageSendSpec{
+		MaxFee: abi.TokenAmount(w.maxWindowPoStGasFee),
+	}
+
+	smsg, err := w.sender.Send(context.Background(), msg, mss, "wdpost")
+	if err != nil {
+		return false, xerrors.Errorf("sending proof message: %w", err)
+	}
+
+	// set message_cid in the wdpost_proofs entry
+
+	_, err = w.db.Exec(context.Background(), `UPDATE wdpost_proofs SET message_cid = $1 WHERE sp_id = $2 AND proving_period_start = $3 AND deadline = $4 AND partition = $5`, smsg.String(), spID, pps, deadline, partition)
+	if err != nil {
+		return true, xerrors.Errorf("updating wdpost_proofs: %w", err)
+	}
+
+	return true, nil
+}
+
+func (w *WdPostSubmitTask) CanAccept(ids []harmonytask.TaskID, engine *harmonytask.TaskEngine) (*harmonytask.TaskID, error) {
+	if len(ids) == 0 {
+		// probably can't happen, but panicking is bad
+		return nil, nil
+	}
+
+	if w.sender == nil {
+		// we can't send messages
+		return nil, nil
+	}
+
+	return &ids[0], nil
+}
+
+func (w *WdPostSubmitTask) TypeDetails() harmonytask.TaskTypeDetails {
+	return harmonytask.TaskTypeDetails{
+		Max:  128,
+		Name: "WdPostSubmit",
+		Cost: resources.Resources{
+			Cpu: 0,
+			Gpu: 0,
+			Ram: 10 << 20,
+		},
+		MaxFailures: 10,
+		Follows:     nil, // ??
+	}
+}
+
+func (w *WdPostSubmitTask) Adder(taskFunc harmonytask.AddTaskFunc) {
+	w.submitPoStTF.Set(taskFunc)
```

Comment:
The API reason to start chain listeners here:
It would enable future harmonytask to start the minimal number of chain listeners for HA. 

I don't know if we will need to give the Daemon a lighter load. For not it's probably fine.